projects/canopy/src/lib/header/header.component.ts

```diff
@@ -20,17 +33,96 @@ import { LgHeaderLogoComponent } from './header-logo/header-logo.component';
     class: 'lg-header',
   },
 })
-export class LgHeaderComponent implements AfterContentInit {
+export class LgHeaderComponent implements AfterContentInit, OnDestroy {
+  private menuTabOutSubscription: Subscription;
+  private menuItemClickedSubscription: Subscription;
+  showResponsiveMenu = false;
+
+  @Output() menuToggled = new EventEmitter<boolean>();
+
+  @ViewChild('menuToggleButton') menuToggleButton: ElementRef;
+  @ContentChild(forwardRef(() => LgPrimaryNavComponent), { read: ElementRef })
+  primaryNavEl: ElementRef;
+  @ContentChild(forwardRef(() => LgPrimaryNavComponent))
+  primaryNav: LgPrimaryNavComponent;
+  @ContentChildren(forwardRef(() => LgPrimaryNavListItemComponent), { descendants: true })
+  navItems: QueryList<LgPrimaryNavListItemComponent>;
   @ContentChildren(forwardRef(() => LgHeaderLogoComponent), {
     descendants: true,
   })
   headerLogos: QueryList<LgHeaderLogoComponent>;
 
+  constructor(private cdr: ChangeDetectorRef) {}
+
+  @HostListener('document:click', [ '$event' ])
+  onDocumentClickout({ target }: MouseEvent): void {
+    if (this.menuToggleButton && this.primaryNavEl) {
+      const { nativeElement: menuToggleButtonEl } = this.menuToggleButton;
+      const { nativeElement: primaryNavEl } = this.primaryNavEl;
+      const isOuterEl =
+        !menuToggleButtonEl.contains(target) && !primaryNavEl.contains(target);
+
+      if (isOuterEl && this.showResponsiveMenu) {
+        this.showResponsiveMenu = false;
+        this.primaryNav.showResponsiveMenu = false;
+        this.cdr.markForCheck();
+      }
+    }
+  }
+
   ngAfterContentInit(): void {
     this.headerLogos.forEach((headerLogo, i) => {
       headerLogo.class = i === 0
         ? 'lg-header-logo__img'
         : 'lg-header-logo__second-img';
     });
+
+    if (this.navItems.length) {
+      this.menuTabOutSubscription = this.navItems.last.tabbedOut.subscribe(
+        (event: KeyboardEvent) => {
+          const isToggleVisible = !!this.menuToggleButton.nativeElement.offsetParent;
+
+          if (isToggleVisible) {
+            event.preventDefault();
+            this.menuToggleButton.nativeElement.focus();
+            this.cdr.markForCheck();
+          }
+        },
+      );
+    }
+
+    const clickedOutputs = this.navItems.map(({ clicked }) => clicked);
+
+    this.menuItemClickedSubscription = merge(...clickedOutputs).subscribe(() => {
```

Comment:
Instead of having the if statement you could use a filer here:
```
    this.menuItemClickedSubscription = merge(...clickedOutputs)
      .filter(() => this.showResponsiveMenu)
      .subscribe(() => {
        this.showResponsiveMenu = false;
        this.primaryNav.showResponsiveMenu = this.showResponsiveMenu;
        this.cdr.markForCheck();
    });
```
Obviously this is up to you, not a blocker at all.

Reply:
Good point, will update